File: apt-integration-tests/src/main/java/me/kpavlov/kt/schema/apt/integration/type/Priority.java
package me.kpavlov.kt.schema.apt.integration.type;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;

/**
 * Simple test model to verify JsonTypeName/JsonProperty name overrides on a Java enum.
 */
@JsonTypeName("PriorityLevel")
public enum Priority {
    @JsonProperty("low")
    LOW,
    @JsonProperty("medium")
    MEDIUM,
    @JsonProperty("high")
    HIGH
}

File: ksp-integration-tests/src/main/kotlin/me/kpavlov/kt/schema/integration/type/Priority.kt
package me.kpavlov.kt.schema.integration.type

import kotlinx.serialization.SerialName
import me.kpavlov.kt.schema.Schema

/**
 * Enum class with SerialName overrides on the class and its entries.
 */
@SerialName("PriorityLevel")
@Schema
enum class Priority {
    @SerialName("p0_critical")
    CRITICAL,

    @SerialName("p1_high")
    HIGH,

    @SerialName("p2_low")
    LOW,
}
